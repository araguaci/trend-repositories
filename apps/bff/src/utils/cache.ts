const LRU = require('lru-cache');

export let cache = null;

export const initCache = () => {
  const options = {
    max: 500,
    ttl: 86400 * 1000, // 1 day
  };

  cache = new LRU(options);
};
