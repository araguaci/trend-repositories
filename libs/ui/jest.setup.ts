import '@testing-library/jest-dom';
jest.mock('next/dist/client/router', () => require('next-router-mock'));
