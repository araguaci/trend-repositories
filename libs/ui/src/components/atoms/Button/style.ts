import styled, { css } from 'styled-components';
import Link from '../Link/Link';
import { LinkProps } from 'next/link';
import { FC } from 'react';
import theme from '../../../theme';

export const StyledButton: FC<
  Partial<LinkProps> & {
    active?: boolean;
    onClick?: () => void;
    color?: keyof typeof theme.colors;
    disabled?: boolean;
  }
> = styled(Link)<{
  active?: boolean;
  color?: keyof typeof theme.colors;
  disabled?: boolean;
}>`
  background-color: ${(props) => props.theme.colors.black_2};
  display: inline-flex;
  align-items: center;
  border-radius: 6px;
  padding: 5px 16px;
  border: 1px solid ${(props) => props.theme.colors.grey_2};

  color: ${(props) =>
    props.color ? props.theme.colors[props.color] : props.theme.colors.white_2};
  font-size: ${(props) => props.theme.sizes.medium};

  :hover {
    text-decoration: none;
    background-color: ${(props) => props.theme.colors.grey_2};
    border-color: ${(props) => props.theme.colors.grey_1};
    transition-duration: 0.1s;
  }

  ${(props) =>
    props.active &&
    css`
      background-color: ${(props) => props.theme.colors.blue_1};

      :hover {
        background-color: ${(props) => props.theme.colors.blue_1};
      }
    `}

  ${(props) =>
    props.disabled &&
    css`
      color: currentColor;
      cursor: not-allowed;
      opacity: 0.5;
      text-decoration: none;
      pointer-events: none;
    `}

  > svg {
    fill: ${(props) =>
      props.color
        ? props.theme.colors[props.color]
        : props.theme.colors.white_2};
    margin-right: 3px;
  }
`;
