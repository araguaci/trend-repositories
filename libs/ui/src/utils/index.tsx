/* istanbul ignore file */
import { ThemeProvider } from 'styled-components';
import theme from '../theme';
import { ReactNode } from 'react';
import { render, act } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import { MockedResponse } from '@apollo/client/testing/core';
import { InMemoryCache } from '@apollo/client/cache';

export const renderWithProvider = (node: ReactNode) => {
  return render(<ThemeProvider theme={theme}>{node}</ThemeProvider>);
};

export const renderWithMockProvider = (
  node: ReactNode,
  cache: InMemoryCache,
  mocks: ReadonlyArray<MockedResponse>
) => {
  return render(
    <ThemeProvider theme={theme}>
      <MockedProvider cache={cache} mocks={mocks}>
        {node}
      </MockedProvider>
    </ThemeProvider>
  );
};

export const renderWithThemeProvider = (node: ReactNode) => {
  return render(<ThemeProvider theme={theme}>{node}</ThemeProvider>);
};

export const waitForResponse = async () =>
  await act(async () => await new Promise((resolve) => setTimeout(resolve, 0)));
export * from '@testing-library/react';
