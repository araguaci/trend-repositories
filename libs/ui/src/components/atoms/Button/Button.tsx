import { StyledButton } from './style';
import { FC } from 'react';
import { LinkProps } from 'next/link';
import theme from '../../../theme';

const Button: FC<
  Partial<LinkProps> & {
    active?: boolean;
    onClick?: () => void;
    color?: keyof typeof theme.colors | undefined;
    disabled?: boolean;
  }
> = ({ children, href, onClick, active, color, disabled, ...props }) => {
  return (
    <StyledButton
      onClick={onClick}
      href={href}
      active={active}
      color={color}
      disabled={disabled}
      {...props}
    >
      {children}
    </StyledButton>
  );
};

export default Button;
