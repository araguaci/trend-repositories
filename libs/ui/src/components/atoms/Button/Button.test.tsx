import { renderWithProvider } from '@trend-repositories/ui/utils';
import Button from './Button';

describe('Button Component', () => {
  it('should be rendered correctly', () => {
    const { asFragment } = renderWithProvider(
      <Button href="https://github.com">linguistic</Button>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
