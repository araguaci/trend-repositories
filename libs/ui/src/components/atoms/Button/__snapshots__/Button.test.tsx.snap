// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should be rendered correctly 1`] = `
<DocumentFragment>
  <a
    class="style__StyledLink-ikww7b-0 fUma-dV style__StyledButton-gn6n84-0 eAVKfN"
    href="https://github.com"
  >
    linguistic
  </a>
</DocumentFragment>
`;
